astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -536,6 +536,67 @@ def divide(self, operand, operand2=None, **kwargs):
         return self._prepare_then_do_arithmetic(np.true_divide, operand,
                                                 operand2, **kwargs)
 
+    def pow(self, exponent, propagate_uncertainties=True, **kwargs):
```

Comment:
Why not add 2 more chars and make the name clearer?
```suggestion
    def power(self, exponent, propagate_uncertainties=True, **kwargs):
```

Reply:
Thanks for the suggestion @pllim.  We could even name it `raise_to_power`.  It's certainly more characters but is much clearer and in keeping with the convention of the other arithmetic operation method names of using a verb.

Reply:
`raise_to_power` might be overkill... but I'll leave that to the subpackage maintainer.

Talking about consistency, do you plan to add these too?

```python
    @sharedmethod
    @format_doc(_arit_doc, name="power", op="**")
```

Reply:
> Talking about consistency, do you plan to add these too?
> 
> ```python
>     @sharedmethod
>     @format_doc(_arit_doc, name="power", op="**")
> ```

I was initially going to.  But then I realised that the `@format_doc` template wasn't suitable as `power` has only one operand while the other methods for which the docstring template was written have two.  Furthermore, the fact that there is only one operand meant that the same WCS, mask and uncertainty handling options aren't valid.  It seemed easier to just write a unique docstring.

As for the `@sharedmethod`, this could be a class method as well.  But due to how the infrastructure is written, the API would have to be written as `result = NDArithmetic.power(exponent, operand)`, with the args in that order.  I am open to this but it is a bit different from the other methods APIs, e.g. `result = NDArithmetic.add(operand1, operand2)`.  So I wanted to get some feedback as to whether this was a good idea.

What do you think @pllim?

Reply:
Ah... okay. Thanks for the clarifications!

Unfortunately, I can't make design decisions in this subpackage as I haven't been involved in its prior development. Perhaps @mwcraig et al. can advise.